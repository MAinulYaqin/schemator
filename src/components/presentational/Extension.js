import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "typeface-roboto";

const Container = styled.div`
  width: 100%;
  padding: 8px 12px;
  box-sizing: border-box;
  cursor: pointer;
  user-select: none;
  background: #333;
  display: flex;

  &:hover {
    background: #383838;
  }
`;

const Image = styled.img`
  width: 48px;
  height: 48px;
  object-fit: cover;
  margin-right: 12px;
`;

const Wrapper = styled.div`
`;

const Name = styled.h4`
  color: #ccc;
  font-size: 14px;
  font-family: Roboto;
  margin: 0px;
  margin-bottom: 4px;
`;

const Author = styled.h5`
  color: #888;
  font-size: 11px;
  font-family: Roboto;
  margin: 0px;
  margin-bottom: 4px;
`;

const Description = styled.p`
  color: #ccc;
  font-family: Roboto;
  font-size: 11px;
  margin: 0px;
`;

const Extension = ({ image, name, author, description, onClick }) => (
  <Container onClick={onClick}>
    <Image src={image} />
    <Wrapper>
      <Name>{name}</Name>
      <Author>{author}</Author>
      <Description>{description}</Description>
    </Wrapper>
  </Container>
);

Extension.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  author: PropTypes.string,
  description: PropTypes.string,
  onClick: PropTypes.func
};

export default Extension;
