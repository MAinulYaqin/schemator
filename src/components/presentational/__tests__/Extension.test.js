import React from "react";
import { shallow } from "enzyme";

import Extension from "../Extension";

describe("Extension", () => {
  const setup = propOverrides => {
    const props = {
      name: "Schema",
      author: "Gattigaga Hayyuta Dewa",
      description: "Create schema in one screen.",
      image: "image.png",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Extension {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders in active", () => {
    const { wrapper } = setup({ isActive: true });

    expect(wrapper.props().isActive).toEqual(true);
  });

  it("should calls 'onClick' while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
