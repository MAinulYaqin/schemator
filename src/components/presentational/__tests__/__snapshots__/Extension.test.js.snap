// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extension should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.img
    src="image.png"
  />
  <styled.div>
    <styled.h4>
      Schema
    </styled.h4>
    <styled.h5>
      Gattigaga Hayyuta Dewa
    </styled.h5>
    <styled.p>
      Create schema in one screen.
    </styled.p>
  </styled.div>
</styled.div>
`;
